src/engine/source/main.cpp

```diff
@@ -1,267 +1,241 @@
-/* Copyright (C) 2015-2022, Wazuh Inc.
- * All rights reserved.
- *
- * This program is free software; you can redistribute it
- * and/or modify it under the terms of the GNU General Public
- * License (version 2) as published by the FSF - Free Software
- * Foundation.
- */
-
-#include <atomic>
-#include <csignal>
-#include <sstream>
+#include <memory>
 #include <string>
-#include <thread>
 #include <vector>
 
-#include <argparse/argparse.hpp>
+#include <CLI/CLI.hpp>
 
-#include <builder.hpp>
-#include <catalog.hpp>
-#include <engineServer.hpp>
-#include <graph.hpp>
-#include <json/json.hpp>
-#include <kvdb/kvdbManager.hpp>
+#include <cmds/cmdGraph.hpp>
+#include <cmds/cmdRun.hpp>
+#include <cmds/cmdTest.hpp>
 #include <logging/logging.hpp>
-#include <profile/profile.hpp>
-#include <protocolHandler.hpp>
-#include <register.hpp>
-// #include <router.hpp>
-#include <hlp/hlp.hpp>
-#include <rxbk/rxFactory.hpp>
-#include <wdb/wdb.hpp>
 
-#define WAIT_DEQUEUE_TIMEOUT_USEC (1 * 1000000)
+#include "base/utils/stringUtils.hpp"
 
-using namespace engineserver;
-
-// Static global variables for handling threads
-static std::atomic<bool> gs_doRun = true;
-static std::vector<std::thread> gs_threadList;
-
-static void sigint_handler(const int signum)
+// Arguments configuration
+namespace args
 {
-    // Inform threads that they must exit
-    gs_doRun = false;
-
-    for (auto& t : gs_threadList)
-    {
-        t.join();
-    };
-
-    exit(0);
+// Subcommand names
+constexpr auto SUBCOMMAND_RUN = "run";
+constexpr auto SUBCOMMAND_LOGTEST = "test";
+constexpr auto SUBCOMMAND_GRAPH = "graph";
+
+// Graph file names
+constexpr auto ENV_DEF_DIR = ".";
+constexpr auto ENV_GRAPH = "env_graph.dot";
+constexpr auto ENV_EXPR_GRAPH = "env_expr_graph.dot";
+
+// Trace all string
+constexpr auto TRACE_ALL = "ALL";
+
+// Arguments
+std::string endpoint;
+std::string file_storage;
+unsigned int queue_size;
+unsigned int threads;
+std::string kvdb_path;
+std::string environment;
+std::string graph_out_dir;
+int protocol_queue;
+std::string protocol_location;
+int debug_level;
+std::string asset_trace;
+int log_level;
+
+void configureSubcommandRun(std::shared_ptr<CLI::App> app)
+{
+    CLI::App* run =
+        app->add_subcommand(args::SUBCOMMAND_RUN, "Run the Wazuh engine module.");
+
+    // Endpoint
+    run->add_option("-e, --endpoint",
+                    args::endpoint,
+                    "Endpoint configuration string. Specifies the endpoint where the "
+                    "engine module will be listening for incoming connections. "
+                    "PROTOCOL_STRING = <protocol>:<ip>:<port>")
+        ->option_text("TEXT:PROTOCOL_STRING REQUIRED")
+        ->required();
+
```

Comment:
If it fills in one line, it is in one line, a new line for description otherwise. This is the default behavior of the lib and seems ok to me. The structure to describe the arguments is the following:
<-f>, <--full_flag> <TYPE>:<SUBTYPE> <[DEFAULT_VALUE] or REQUIRED> <description>

I don't see where the confusion is.